### ISSUE TITLE
spin test with arguments does not work when `spin test` works alone

### ISSUE THREAD

**Issue by adam2392** [scientific-python/spin#208]
I am running into some weird behavior when upgrading spin recently. I am running spin on GH actions and locally: https://github.com/neurodata/scikit-tree/actions/runs/9441764126/job/26003160931

```
(sktree) (base) adam2392@arm64-apple-darwin20 scikit-tree % ./spin test -t ./sktree --coverage --gcov -- -m "slowtest"
Invoking `build` prior to running tests:
$ meson setup build --prefix=/usr -Db_coverage=true --reconfigure
The Meson build system
Version: 1.4.0
Source dir: /Users/adam2392/Documents/scikit-tree
Build dir: /Users/adam2392/Documents/scikit-tree/build
Build type: native build
Project name: scikit-tree
Project version: 0.8.0dev0
C compiler for the host machine: arm64-apple-darwin20.0.0-clang (clang 15.0.7 "clang version 15.0.7")
C linker for the host machine: arm64-apple-darwin20.0.0-clang ld64 609
C++ compiler for the host machine: arm64-apple-darwin20.0.0-clang++ (clang 15.0.7 "clang version 15.0.7")
C++ linker for the host machine: arm64-apple-darwin20.0.0-clang++ ld64 609
Cython compiler for the host machine: cython (cython 3.0.10)
Host machine cpu family: aarch64
Host machine cpu: aarch64
Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
Compiler for C supports arguments -Wno-unused-function: YES (cached)
Compiler for C supports arguments -Wno-conversion: YES (cached)
Compiler for C supports arguments -Wno-misleading-indentation: YES (cached)
Compiler for C supports arguments -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION: YES (cached)
Library m found: YES
Program cython found: YES (/Users/adam2392/miniforge3/envs/sktree/bin/cython)
Program python3 found: YES (/Users/adam2392/miniforge3/envs/sktree/bin/python3.9)
Message: /Users/adam2392/miniforge3/envs/sktree/bin/python3.9
Message: /usr/lib/python3.9/site-packages/
Dependency OpenMP found: YES 5.0 (cached)
Build targets in project: 19

scikit-tree 0.8.0dev0

  User defined options
    prefix    : /usr
    b_coverage: true

Found ninja-1.11.1.git.kitware.jobserver-1 at /Users/adam2392/miniforge3/envs/sktree/bin/ninja
WARNING: Need gcovr or lcov/genhtml to generate any coverage reports                          
Cleaning... 0 files.                                                                          
$ meson compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /Users/adam2392/miniforge3/envs/sktree/bin/ninja -C /Users/adam2392/Documents/scikit-tree/build
ninja: Entering directory `/Users/adam2392/Documents/scikit-tree/build'
… <some build information that I removed>
ld: warning: -pie being ignored. It is only used when linking a main executable
$ meson install --only-changed -C build --destdir ../build-install
$ export PYTHONPATH="/Users/adam2392/Documents/scikit-tree/build-install/usr/lib/python3.9/site-packages"
$ export PYTHONPATH="/Users/adam2392/Documents/scikit-tree/build-install/usr/lib/python3.9/site-packages"
$ /Users/adam2392/miniforge3/envs/sktree/bin/python -c 'import sys; del sys.path[0]; import sktree'
$ pytest --pyargs ./sktree -m slowtest --cov-report=term --cov-report=html:/Users/adam2392/Documents/scikit-tree/build/coverage/ --cov=sktree
ImportError while loading conftest '/Users/adam2392/Documents/scikit-tree/sktree/conftest.py'.
sktree/__init__.py:67: in <module>
    raise ImportError(e) from e
E   ImportError: No module named 'sktree._lib.sklearn.__check_build._check_build'
E   ___________________________________________________________________________
E   Contents of /Users/adam2392/Documents/scikit-tree/sktree/_lib/sklearn/__check_build:
E   __init__.py               __pycache__               meson.build
E   _check_build.pyx
E   ___________________________________________________________________________
E   It seems that scikit-learn has not been built correctly.
E
E   If you have installed scikit-learn from source, please do not forget
E   to build the package before using it: run `python setup.py install` or
E   `make` in the source directory.
E
E   If you have used an installer, please check that it is suited for your
E   Python version, your operating system and your platform.
```

However, if I do `./spin install` and then run things with `pytest` locally it works fine. I"m wondering if there is some issue with test discovery, or not using the `pytest.ini` options in pyproject.toml when using `spin test`?

**Comment by ngoldbaum** [comment]
https://github.com/numpy/numpy/issues/26525 is related. I'm hitting this on my mac dev environment running `spin test` in the numpy repo. Happy to help out with debugging, for now I'm downgrading to spin 0.8...

**Comment by stefanv** [comment]
Unconditional pyargs was probably a mistake. I'll add some additional logic to better detect file-based tests.

**Comment by stefanv** [comment]
Closed by #213
